st3/sublime_lib/view_utils.py

```diff
@@ -0,0 +1,53 @@
+from .syntax import get_syntax_for_scope
+from .encodings import to_sublime
+
+
+def new_view(window, **kwargs):
+    if 'scope' in kwargs and 'syntax' in kwargs:
```

Comment:
Hm, you have a point. My other though was to wrap the set options call ina `try...except` but then re-raise after closing the view.

Reply:
I'd rather not open up the view at all if we know it's going to error. The available approach for closing a view is kind of hacky in the first place. The way I'm thinking of it, if you've passed invalid arguments to `set_view_options`, then you've passed invalid arguments to `new_view`, and it should fail immediately in `new_view`. With that in mind, it would probably be best to factor out the arguments validation into a separate function, then invoke it from both `new_view` and from the `OutputPanel` constructor.